### ISSUE TITLE
subdomain with hyphen crashes nginx

### ISSUE THREAD

**Issue by sgotre** [laravel/valet#808]
Laravel Valet 2.3.3
Added a subdomain containing more than one hyphen.
This crashes nginx. 

Works:
```
$  valet link --secure subdomain-minus.mydomain
```

nginx can not start anymore
```
$  valet link --secure subdomain-minus-test-test.mydomain
```

You can get valet back working by deleting this file: 
```
 /Users/{MY_USER_ACCOUNT}/.config/valet/Nginx/subdomain-minus-test-test.mydomain.{YOURTLD}
```

**Comment by drbyte** [comment]
Using `dev-master` (which isn't significantly different presently from v`2.3.3`), I can't replicate your symptoms, at least not using your exact example.
If your "actual" subdomain+domain is very very long, there's a known issue with that being problematic (I don't recall it being related to hyphens at all though). 
But the length in your example doesn't cause any problem for me:

<img width="1155" alt="Screen Shot 2019-08-15 at 9 12 10 AM" src="https://user-images.githubusercontent.com/404472/63096806-d987a800-bf3c-11e9-84df-71161ba967ab.png">
<img width="1174" alt="Screen Shot 2019-08-15 at 9 08 03 AM" src="https://user-images.githubusercontent.com/404472/63096814-dc829880-bf3c-11e9-84ca-39a1acaeb1e3.png">

**Comment by sgotre** [comment]
@drbyte Thanks for you fast reply
yes, i had issues with longer domains. 
Just retried with long domain names. You are right. It is not related to hyphens, but to domain length. 

**TLDR: Total domain length has a limit of 43 characters. Valet is not throwing an error**

You can try it yourself (i tested on v2.3.3)

valet link --secure 20characters12345678  
 => working

valet link --secure 30characters123456789123456789
 => working

valet link --secure  36characters123456789123456789123456
=> working

valet link --secure  37characters1234567891234567891234567
=> working

valet link --secure  38characters12345678912345678912345678
=> failed

All longer domain names fail: 
valet link --secure  40characters1234567891234567891234567890
=> failing

**The domain length has a limit of 37 characters when using TLD .test**

When switching the TLD to .a:

valet link --secure 40characters1234567891234567891234567890
=> works

valet link --secure 41characters12345678912345678912345678901
 => fails

So the total domain length has a limit of 43 characters (including .tld)
Maybe valet should throw an error if a domain is longer than that.

**Comment by drbyte** [comment]
@sgotre Thanks for posting that research.

Turns out there's an easy fix:
Add a new file:
`~/.config/valet/Nginx/allow_longer_domain_names.conf`
containing:
```
# Use 128 here since using 32 or 64 may limit domain names to shorter than 43 characters in Valet
server_names_hash_bucket_size 128;
```

**Comment by sgotre** [comment]
@drbyte thank you for this fix! Closing this issue in favor of #543
